systems/cryptography/caesar-cipher/Java/shakeel.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * @author: shakeelsamsu
+ * https://github.com/shakeelsamsu
+ * Oct 6 18 4:49:02 PM
+ * An implementation of the Caesar Cipher
+ * Input: A String of lower case letters, followed by
+ * an Integer between 0-25 denoting the required shift.
+ * Output: The encrypted message
+ */
+
+import java.io.*;
+import java.util.*;
+
+public class shakeel {
+    public static void main(String[] args) throws IOException  {
+        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+        PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
+        StringTokenizer st = new StringTokenizer(in.readLine());
+
+        ArrayList<Character> letters = new ArrayList<Character>();
+        for(int i = 0; i < 26; i++) letters.add((char) (i + 97));
+        String input = st.nextToken();
+        char[] arr = input.toCharArray();
+        int shift = Integer.parseInt(st.nextToken());
+        for(int j = 0; j < arr.length; j++) {
+            char originalChar = arr[j];
+            int shiftedNum = letters.indexOf(originalChar) + shift; 
```

Comment:
This will increase the time complexity by a factor of `n`. Can't the index be directly found by subtracting 97 from ASCII value of `originalChar`?

Reply:
I believe so, but using this adds to the point of having the HashMap. Changing it now seems unnecessary, but I could incorporate that approach instead if it's preferred.